apps/rush-lib/src/logic/test/ProjectChangeAnalyzer.test.ts

```diff
@@ -33,6 +33,9 @@ describe(ProjectChangeAnalyzer.name, () => {
       },
       findProjectForPosixRelativePath(path: string): object | undefined {
         return projects.find((project) => path.startsWith(project.projectRelativeFolder));
+      },
+      getProjectByName(name: string): object | undefined {
+        return projects.find((project) => project.packageName === name);
```

Comment:
Seems like it would be more efficient to just have a lazily-initialized map of projects by name.

Reply:
I'm gonna leave as-is since it's just unit-test code and there's existing examples using this pattern.

Reply:
Looks like it's being used here: https://github.com/microsoft/rushstack/pull/2814/files#diff-a97c75b62243d6b88ead5b45905024ed5352dbf21bd23460511267ad924548c6R67

Reply:
Ohh sorry, I thought this was in RushConfiguration. Nevermind.